src/Compiler/Microsoft.AspNetCore.Razor.Language/test/HtmlConventionsTest.cs

```diff
@@ -21,7 +21,7 @@ public static TheoryData HtmlConversionData
                     { "CAPSOnOUTSIDE", "caps-on-outside" },
                     { "ALLCAPS", "allcaps" },
                     { "One1Two2Three3", "one1-two2-three3" },
-                    { "ONE1TWO2THREE3", "one1two2three3" },
+                    { "ONE1TWO2THREE3", "one1-two2-three3" },
```

Comment:
This is the change in behaviour which I deemed to be unimportant.

Reply:
Another test failed with this change, and looking at it, it would be a user facing regression I think, so I'm going to expand the algorithm to match.